storage/innobase/row/row0import.cc

```diff
@@ -4279,6 +4527,50 @@ static void row_import_autoinc(dict_table_t *table, row_prebuilt_t *prebuilt,
   }
 }
 
+/**
+1) Update the position of the columns and
+2) Insert the hidden fts doc id in the sys columns table
+3) Insert the hidden fts doc id in the sys indexes and
+sys_fields table
+@param  table   table to be imported
+@param  fts_pos position of fts doc id column
+@param  trx     transaction
+@return DB_SUCCESS or error code */
+static
+dberr_t innodb_insert_hidden_fts_col(dict_table_t* table,
+                                     ulint  fts_pos,
+                                     trx_t* trx)
+{
+  dict_index_t* fts_idx=
+    dict_table_get_index_on_name(table, FTS_DOC_ID_INDEX_NAME);
+  if (!fts_idx) return DB_ERROR;
+
+  pars_info_t *info= pars_info_create();
+  pars_info_add_ull_literal(info, "id", table->id);
+  dict_hdr_get_new_id(NULL, &fts_idx->id, NULL);
+  pars_info_add_ull_literal(info, "idx_id", fts_idx->id);
+  pars_info_add_int4_literal(info, "pos", fts_pos);
+  pars_info_add_int4_literal(info, "space", fts_idx->table->space_id);
+  pars_info_add_int4_literal(info, "page_no", fts_idx->page);
+
+  return que_eval_sql(info,
+                      "PROCEDURE ADD_FTS_COL () IS\n"
+                      "BEGIN\n"
+                      "UPDATE SYS_COLUMNS SET\n"
+                      "POS = POS + 1\n"
+                      "WHERE TABLE_ID = :id AND POS >= :pos;\n"
```

Comment:
Do we need this? Shouldn’t the hidden `FTS_DOC_ID` (which we are adding) always be the last column?

I can imagine that we might need this because of virtual columns? If that is the case, then we should update `SYS_VIRTUAL` as well.